src/connectors/scenes/FioConnectWalletConnector.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import { connect } from 'react-redux'
+
+import type { FioConnectWalletStateProps as StateProps } from '../../components/scenes/FioConnectWalletScene'
+import { FioConnectWalletScene } from '../../components/scenes/FioConnectWalletScene'
+import { isPubAddressNotConnected } from '../../modules/FioAddress/util'
+import { getFioWallets, getWallets } from '../../modules/UI/selectors'
+import type { State } from '../../types/reduxTypes'
+
+const mapStateToProps = (state: State, ownProps): StateProps => {
+  const wallets = getWallets(state)
+  const fioWallets = getFioWallets(state)
+  const connectedPubAddresses = state.ui.fio.connectedPubAddresses[ownProps.fioAddressName]
+  const notConnectedWallets = {}
+
+  if (!connectedPubAddresses) return { wallets: {}, fioWallets, loading: true }
+
+  for (const walletKey: string in wallets) {
```

Comment:
move this logic into the component

Reply:
left here for now (moved to the util, but set object here), because componentDidMount there is tricky to check if wallets were updated

Reply:
Do not use component did mount. Instead use component will update

Reply:
sorry, I mean componentDidUpdate not didMound. but anyway componentWillUpdate is a legacy method, better to use some workaround with `getDerivedStateFromProps` and setting previous props in state, but it also tricky, so I think easier to leave it here